x/staking/src/types/domain.rs

```diff
@@ -235,7 +244,8 @@ impl Validator {
         // calculate the shares to issue
         let issues_shares = if self.delegator_shares.is_zero() {
             // the first delegation to a validator sets the exchange rate to one
-            Decimal256::new(amount)
+            Decimal256::from_atomics(amount, 0)
+                .expect("Conversion won't fail. The number of decimal places is 0")
```

Comment:
I don't think it's true that this conversion can't fail see https://docs.rs/cosmwasm-std/latest/cosmwasm_std/struct.Decimal256.html "The greatest possible value that can be represented is 115792089237316195423570985008687907853269984665640564039457.584007913129639935 (which is (2^256 - 1) / 10^18)"